### PR TITLE
Fixes to handling JSON types in outputs

### PR DESCRIPTION
**Pull request by DonJayamanne**
For #6017

### CODE REVIEW

File: src/client/datascience/notebook/helpers/helpers.ts [hunk 1]
```diff
@@ -560,7 +560,16 @@ function translateDisplayDataOutput(
     for (const key in data) {
         // Add metadata to all (its the same)
         // We can optionally remove metadata that belongs to other mime types (feels like over optimization, hence not doing that).
-        items.push(new NotebookCellOutputItem(convertJupyterOutputToBuffer(key, data[key]), key, metadata));
+        const value = data[key];
+        let itemMetadata = metadata;
+        if (typeof value === 'object' && value !== null && !Array.isArray(value)) {
+            itemMetadata = JSON.parse(JSON.stringify(metadata));
```
Comment (joyceerhl) [inline src/client/datascience/notebook/helpers/helpers.ts:566]: Just for my understanding, why is this part necessary?

Comment (DonJayamanne) [reply src/client/datascience/notebook/helpers/helpers.ts:566]: Cloning it, will add comments

Comment (DonJayamanne) [reply src/client/datascience/notebook/helpers/helpers.ts:566]: Each output item can have its own metqdata, i don't want all output items to end up with a flag `__isJson` that would be wrong

**Review by joyceerhl — APPROVED** [review]
